### PR TITLE
ENH add osx version virtual package

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds a virtual package for the OSX version.

### CODE REVIEW

**Comment by beckermr** [comment]
I cannot get the test to pass due to some weird interaction of patch with the decorated attributes. I'm running out of motivation here.

**Comment by msarahan** [comment]
This seems fine.  The one test fail I see is something subtle in the solver that pops up now and again.  It's not important.

**Comment by msarahan** [comment]
@beckermr as the annoying bot has indicated, we don't seem to have a signed CLA from you.  Can you please sign the CLA linked to in the bot's messages?  I'll merge once we get that.

**Comment by beckermr** [comment]
I am working with my local lawyers to sign it. Sorry for the delay. Being at a national lab is annoying sometimes with the red tape

**Comment by msarahan** [comment]
Understood!  Sorry to make you go through that trouble, but thankful that you are enduring.

**Comment by beckermr** [comment]
I signed the CLA. Hopefully this can be merged in the coming week!

**Comment by isuruf** [comment]
@beckermr, the test for glibc was skipped for non unix in https://github.com/conda/conda/pull/9358. Maybe we can do the same here?

**Comment by beckermr** [comment]
Ahhhh man. Merge conflicts! :(

Yeah I am happy to skip the test, but this is not up to me.

**Comment by beckermr** [comment]
Alright. Merge conflicts fixed. I'd like to merge this PR soon so it doesn't go stale again. I signed the CLA but the bot still thinks I have not. Please let me know if I need to do something further @msarahan !

File: tests/core/test_index.py [hunk 1]
```diff
@@ -54,6 +54,17 @@ def test_supplement_index_with_system_cuda():
     assert cuda_pkg.package_type == PackageType.VIRTUAL_SYSTEM
 
 
+@pytest.mark.skipif(on_win or on_linux, reason="osx-only test")
```
Comment (isuruf) [inline tests/core/test_index.py]:
```suggestion
@pytest.mark.skipif(not on_mac, reason="osx-only test")
```

**Comment by msarahan** [comment]
@cla-bot check

**Comment by beckermr** [comment]
I added an `on_linux` flag to common.compat. Feel free to keep or remove.

**Comment by msarahan** [comment]
@jjhelmus please merge when you're ready and include this in 4.8.0

**Comment by jjhelmus** [comment]
@beckermr Thanks for putting this together.  We are planning on tagging a 4.8.0 release in the next few days so this will be available for use soon.

**Pull request merged**